### PR TITLE
Feature: Password protection for server.

### LINKED ISSUE #1649: Password protection for server
Clearly a bigger task, consisting of:
- [x] Mod schema to allow password for `:presence/hello` event
- [x] Mod `common-events/build-presence-hello-event` to accept `password`
- [x] Mod configuration (and env) to allow specifying server password
- [x] Mod `athens.self-hosted.web.presence/hello-handler` to reject users with wrong password
- [x] Mod Client to allow editing password
- [x] Mod client to send password when configuration has it
- [x] Mod Client to handle `:presence/hello` rejections

### PR DESCRIPTION
**Pull request by neotyk**
Closes #1649.

Tasks are listed in the issue above.

### CODE REVIEW

File: src/clj/athens/self_hosted/clients.clj [hunk 1]
```diff
@@ -69,7 +69,7 @@
 ;; Client management API
 (defn add-client!
   ([channel]
-   (add-client! channel true))
+   (add-client! channel false))
```
Comment (filipesilva) [inline src/clj/athens/self_hosted/clients.clj:72]: What's going on here lol?

File: src/cljs/athens/electron/utils.cljs [hunk 2]
```diff
@@ -61,12 +61,13 @@
...
-   :name   name
-   :id     url
-   :url    url
-   :ws-url (str "ws://" url "/ws")})
+  [name url password]
+  {:type     :self-hosted
+   :name     name
+   :id       url
+   :url      url
+   :password password
```
Comment (filipesilva) [inline src/cljs/athens/electron/utils.cljs:69]: We'll need to review the storage and transmission of sensitive information like passwords and tokens at some point. Out of scope for now, just wanted to mention it.

**Review by filipesilva — APPROVED** [review]

File: src/clj/athens/self_hosted/clients.clj (hunk 1 above)
Comment (neotyk) [reply src/clj/athens/self_hosted/clients.clj:72]: that's a client that didn't yet introduce itself.

File: src/cljs/athens/electron/utils.cljs (hunk 2 above)
Comment (neotyk) [reply src/cljs/athens/electron/utils.cljs:69]: Indeed

**Pull request merged**